tests/mocha/serializer_test.js

```diff
@@ -0,0 +1,1754 @@
+    'Japanese',
+    '<xml xmlns="https://developers.google.com/blockly/xml">' +
+    '<block type="procedures_defreturn" id="id******************" x="42" y="42">' +
+    '<field name="NAME">あなたは好奇心旺盛な猫です</field>' +
+    '</block>' +
+    '</xml>');
+Serializer.Mutations.Procedure.Names.Zalgo = new SerializerTestCase(
+    'Zalgo',
+    '<xml xmlns="https://developers.google.com/blockly/xml">' +
+    '<block type="procedures_defreturn" id="id******************" x="42" y="42">' +
+    '<field name="NAME">z̴̪͈̲̜͕̽̈̀͒͂̓̋̉̍a̸̧̧̜̻̘̤̫̱̲͎̞̻͆̋ļ̸̛̖̜̳͚̖͔̟̈́͂̉̀͑̑͑̎ǵ̸̫̳̽̐̃̑̚̕o̶͇̫͔̮̼̭͕̹̘̬͋̀͆̂̇̋͊̒̽</field>' +
+    '</block>' +
+    '</xml>');
+Serializer.Mutations.Procedure.Names.ControlChars = new SerializerTestCase(
+    'ControlChars',
+    '<xml xmlns="https://developers.google.com/blockly/xml">' +
+    '<block type="procedures_defreturn" id="id******************" x="42" y="42">' +
+    '<field name="NAME">&#x01;&#xa1;</field>' +
+    '</block>' +
+    '</xml>');
+Serializer.Mutations.Procedure.Names.testCases = [
+  Serializer.Mutations.Procedure.Names.Symbols,
+  Serializer.Mutations.Procedure.Names.EscapedSymbols,
+  Serializer.Mutations.Procedure.Names.SingleQuotes,
+  Serializer.Mutations.Procedure.Names.DoubleQuotes,
+  Serializer.Mutations.Procedure.Names.Numbers,
+  Serializer.Mutations.Procedure.Names.Emoji,
+  Serializer.Mutations.Procedure.Names.Russian,
+  Serializer.Mutations.Procedure.Names.Japanese,
+  Serializer.Mutations.Procedure.Names.Zalgo,
+  // TODO: Uncoment once #4945 is merged.
+  // Serializer.Mutations.Procedure.Names.ControlChars,
+];
+
+Serializer.Mutations.Procedure.testSuites = [
+  Serializer.Mutations.Procedure.Names,
+];
+
+Serializer.Mutations.testSuites = [
+  Serializer.Mutations.ControlsIf,
+  Serializer.Mutations.ListCreate,
+  Serializer.Mutations.Procedure,
+];
+
+Serializer.testSuites = [
+  Serializer.Attributes,
+  Serializer.Fields,
+  Serializer.Icons,
+  Serializer.Connections,
+  Serializer.Mutations,
+];
+
+var runSerializerTestSuite = (serializer, deserializer, testSuite) => {
+
+  const createTestFunction = function(test) {
+    return function() {
+      Blockly.Xml.domToWorkspace(
+          Blockly.Xml.textToDom(test.xml), this.workspace);
+      if (serializer && deserializer) {
+        // do custom serialization and deserialization.
+      }
+      var newXml = Blockly.Xml.workspaceToDom(this.workspace);
+      chai.assert.equal(Blockly.Xml.domToText(newXml), test.xml);
+    };
+  };
+
+  const createSuiteOrTestFunction = function(suiteOrTest) {
+    if (suiteOrTest instanceof testHelpers.TestSuite) {
+      return createSuiteOrTestFunction;
+    }
+    return createTestFunction(suiteOrTest);
+  };
+
+  let suiteCall = (testSuite.skip ? suite.skip : suite);
+  suiteCall = (testSuite.only ? suite.only : suiteCall);
+
+  suiteCall(testSuite.title, function() {
+    setup(function() {
+      this.workspace = new Blockly.Workspace();
```

Comment:
Should this be calling `sharedTestSetup` @moniika  ?

Reply:
Yeah, I would add `sharedTestSetup` and `sharedTestTeardown`.
Although it's not in Samples, it's not much work to add a call here and It's helprf me catch bugs with tests before, so it could be useful.